src/bin/vip-import-sql.js

```diff
@@ -192,13 +214,154 @@ const examples = [
 	},
 ];
 
+const promptToContinue = async ( {
+	launched,
+	formattedEnvironment,
+	track,
+	domain,
+} ): Promise<void> => {
+	console.log();
+	const promptToMatch = domain.toUpperCase();
+	const promptResponse = await prompt( {
+		type: 'input',
+		name: 'confirmedDomain',
+		message: `You are about to import the above tables into a ${
+			launched ? 'launched' : 'un-launched'
+		} ${ formattedEnvironment } site ${ chalk.yellow( domain ) }.\nType '${ chalk.yellow(
+			promptToMatch
+		) }' (without the quotes) to continue:\n`,
+	} );
+
+	if ( promptResponse.confirmedDomain !== promptToMatch ) {
+		await track( 'import_sql_unexpected_tables' );
+		exit.withError( 'The input did not match the expected environment label. Import aborted.' );
+	}
+};
+
+export type validateAndGetTableNamesInputType = {
+	skipValidate: boolean,
+	appId: number,
+	envId: number,
+	fileNameToUpload: string,
+};
+
+export async function validateAndGetTableNames( {
+	skipValidate,
+	appId,
+	envId,
+	fileNameToUpload,
+}: validateAndGetTableNamesInputType ): Promise<Array<string>> {
+	const validations = [ staticSqlValidations, siteTypeValidations ];
+	let tableNamesInSqlFile = [];
+	if ( skipValidate ) {
+		console.log( 'Skipping SQL file validation.' );
+	} else {
+		try {
+			await fileLineValidations( appId, envId, fileNameToUpload, validations );
+		} catch ( validateErr ) {
+			console.log( '' );
+			exit.withError( `${ validateErr.message }
+
+If you are confident the file does not contain unsupported statements, you can retry the command with the ${ chalk.yellow(
+		'--skip-validate'
+	) } option.
+` );
+		}
+		// this can only be called after static validation of the SQL file
+		tableNamesInSqlFile = getTableNames();
+	}
+	return tableNamesInSqlFile;
```

Comment:
... & we can drop an indention level because the `return` obviates the `else` case:

```suggestion
	if ( skipValidate ) {
		console.log( 'Skipping SQL file validation.' );
		return [];
	}
	try {
		await fileLineValidations( appId, envId, fileNameToUpload, validations );
	} catch ( validateErr ) {
		console.log( '' );
		exit.withError( `${ validateErr.message }
If you are confident the file does not contain unsupported statements, you can retry the command with the ${ chalk.yellow( '--skip-validate' ) } option.
` );
	}
	// this can only be called after static validation of the SQL file
	return getTableNames();
```